lib/middleware/redirects.js

```diff
@@ -30,14 +30,22 @@ function addQuery(url, qs) {
   return url;
 }
 
-var Redirect = function(source, destination, type) {
+var Redirect = function(glob, pattern, destination, type) {
   this.type = type || 301;
-  this.source = slasher(source);
+  this.glob = slasher(glob);
+  this.pattern = pattern;
   this.destination = destination;
 
   if (this.destination.match(/(?:^|\/):/)) {
     this.captureKeys = [];
```

Comment:
OK, so this is kind of the Original Sin for this file; pathToRegexp (because it's actually lexxing the entire glob as it goes around, and because globs are a subset of regexps it can get away with this) knows the full set of captures when it's done with the glob and puts it into `captureKeys`

like if you call PathToRegexp on "/foo/:bar/:baz", captureKeys will be ["bar", "baz"] afterwards.

The problem is that isn't (and can't) be a similar precomputed list of capture keys we're interested in for general-case regexps. The rest of the file is basically working around that, and I'm not like...100% sure of it yet.

Reply:
@mbleigh to comment

Reply:
Since RE matchers are likely to remain fairly uncommon, let's just get something that works landed and we can iterate and think about superstatic as a whole as we go.